File: tmp/5.最长回文子串.cpp
/*
 * @lc app=leetcode.cn id=5 lang=cpp
 *
 * [5] 最长回文子串
 */
#include <string>
#include <vector>
using namespace std;

// @lc code=start
class Solution
{
public:
    int reverse(string s, int idx)
    {
        return s[s.length() - 1 - idx];
    }
    string longestPalindrome(string s)
    {
        int slength = s.length();

        //
        int lenPalindrome[slength][slength] = {0};

        int path[slength][slength] = {0};

        // 初始化第0行
        for (int i = 0; i < slength; i++)
        {
            if (i == 0)
            {
                if (s[0] == reverse(s, 0))
                {
                    lenPalindrome[0][0] = 1;
                    
                }
            }
        }

        // 初始化第0列
        for (int i = 0; i < slength; i++)
        {
        }
    }
};
// @lc code=end

File: tmp/5.最长回文子串.py
#
# @lc app=leetcode.cn id=5 lang=python3
#
# [5] 最长回文子串
#


# @lc code=start
class Solution:
    def reverse(self, s: str, idx: int):
        return s[len(s) - 1 - idx]

    def longestPalindrome(self, s: str) -> str:
        str1 = s
        str2 = list(reversed(s))
        str_len = len(s)

        max_len_sub = []
        


# @lc code=end
